selfdrive/test/process_replay/regen.py

```diff
@@ -268,11 +270,18 @@ def regen_and_save(route, sidx, upload=False, use_route_meta=False):
     lr = LogReader(r.log_paths()[args.seg])
     fr = FrameReader(r.camera_paths()[args.seg])
   else:
-    lr = LogReader(f"cd:/{route.replace('|', '/')}/{sidx}/rlog")
+    lr = LogReader(f"cd:/{route.replace('|', '/')}/{sidx}/rlog.bz2")
     fr = FrameReader(f"cd:/{route.replace('|', '/')}/{sidx}/fcamera.hevc")
   rpath = regen_segment(lr, {'roadCameraState': fr})
 
-  lr = LogReader(os.path.join(rpath, 'rlog2'))
+  # compress raw rlog before uploading
+  with open(os.path.join(rpath, "rlog"), "rb") as f:
+    data = bz2.compress(f.read())
+  with open(os.path.join(rpath, "rlog.bz2"), "wb") as f:
+    f.write(data)
+  os.remove(os.path.join(rpath, "rlog"))
+
+  lr = LogReader(os.path.join(rpath, 'rlog.bz2'))
```

Comment:
we could instead regenerate all the routes uncompressed, but then we upload and store 40MB per segment instead of 10-13, which also takes much longer to upload

Reply:
Compressing before uploading is fine!